src/toxcore/dht/client.rs

```diff
@@ -135,35 +135,40 @@ impl Client {
                 IfAddr::V4(ref addr) => addr.broadcast,
                 _ => None,
         })
-        .map(|ipv4|
-            IpAddr::V4(ipv4)
-        ).collect()
+        .map(|addr|
+            IpAddr::V4(addr)
+        )
+        .collect()
     }
-    // get broadcast addresses for Ipv4 and Ipv6
-    fn get_broadcast_addresses() -> Vec<IpAddr> {
-        let mut res = Vec::new();
-        let ipv4s = Client::get_ipv4_broadcast_addrs();
-
-        res.extend(ipv4s.iter());
-
+    /// send LanDiscovery packets to broadcast addresses when dht_node runs as Ipv4 mode
+    pub fn send_lan_discovery_ipv4(&self) -> IoFuture<()> {
+        let mut ip_addrs = Client::get_ipv4_broadcast_addrs();
         // Ipv4 global broadcast address
-        res.push(
-            ("255.255.255.255").parse().unwrap()
+        ip_addrs.push(
+            "255.255.255.255".parse().unwrap()
         );
-        // Ipv6 broadcast address
-        res.push(
-            ("FF02::1").parse().unwrap()
+        let lan_packet = DhtPacket::LanDiscovery(LanDiscovery {
+            pk: self.pk,
+        });
+        let lan_sender = ip_addrs.iter().map(|&addr|
+            self.send_to(SocketAddr::new(addr, 33445), lan_packet.clone()) // 33445 is default port for tox
         );
-        res
+
+        let lan_stream = stream::futures_unordered(lan_sender).then(|_| Ok(()));
+        Box::new(lan_stream.for_each(|()| Ok(())))
     }
-    /// send LanDiscovery packets to broadcast addresses
-    pub fn send_lan_discovery(&self) -> IoFuture<()> {
-        let ip_addrs = Client::get_broadcast_addresses();
+    /// send LanDiscovery packets to broadcast addresses when dht_node runs as Ipv6 mode
+    pub fn send_lan_discovery_ipv6(&self) -> IoFuture<()> {
+        let mut ip_addrs = Client::get_ipv4_broadcast_addrs();
+        // Ipv6 broadcast address
+        ip_addrs.push(
+            "::1".parse().unwrap() // it should be FF02::1, but for now, my LAN config has no route to address of FF02::1
```

Comment:
c-toxcore in case of IPv6 sends packet to one more address - 255.255.255.255 mapped to IPv6. Maybe we should add it?